app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt

```diff
@@ -0,0 +1,177 @@
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.github.se.eventradar.R
+import com.github.se.eventradar.model.EventsOverviewViewModel
+import com.github.se.eventradar.ui.BottomNavigationMenu
+import com.github.se.eventradar.ui.component.*
+import com.github.se.eventradar.ui.map.EventMap
+import com.github.se.eventradar.ui.navigation.NavigationActions
+import com.github.se.eventradar.ui.navigation.Route
+import com.github.se.eventradar.ui.navigation.TOP_LEVEL_DESTINATIONS
+import com.github.se.eventradar.ui.navigation.getTopLevelDestination
+import com.github.se.eventradar.util.toast
+
+@Composable
+fun HostingScreen(
+    viewModel: EventsOverviewViewModel = viewModel(),
+    navigationActions: NavigationActions
+) {
+  val uiState by viewModel.uiState.collectAsState()
+  LaunchedEffect(key1 = uiState.eventList) { viewModel.getEvents() }
+  var viewMapOrListIndex by remember { mutableIntStateOf(0) }
+  ConstraintLayout(modifier = Modifier.fillMaxSize().testTag("hostingScreen")) {
+    val (logo, title, divider, eventList, eventMap, bottomNav, buttons) = createRefs()
+    Logo(
+        modifier =
+            Modifier.fillMaxWidth()
+                .constrainAs(logo) {
+                  top.linkTo(parent.top, margin = 32.dp)
+                  start.linkTo(parent.start, margin = 16.dp)
+                }
+                .testTag("logo"))
+    Title(
+        modifier =
+            Modifier.testTag("myHostedEventsTitle")
+                .constrainAs(
+                    title,
+                    {
+                      top.linkTo(logo.bottom, margin = 16.dp)
+                      start.linkTo(parent.start)
+                      end.linkTo(parent.end)
+                    }))
+    HorizontalDivider(
+        modifier = Modifier.constrainAs(divider, { top.linkTo(title.bottom, margin = 10.dp) }))
+    if (viewMapOrListIndex == 0) {
+      EventList(
+          uiState.eventList
```

Comment:
I see your comment, but `filteredEvent` is meant to filter the events on the homescreen, not the hosting screen.